language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureHelpUtil.java

```diff
@@ -92,21 +92,35 @@ private SignatureHelpUtil() {
         SignatureInfoModel signatureInfoModel = getSignatureInfoModel(functionSymbol.get(), context);
 
         // Override label for 'new' constructor
-        String label = functionSymbol.get().name();
-        int initIndex = label.indexOf(INIT_SYMBOL);
-        if (initIndex > -1) {
-            label = "new " + label.substring(0, initIndex);
-        }
+        int initIndex = functionSymbol.get().name().indexOf(INIT_SYMBOL);
+        StringBuilder labelBuilder = initIndex > -1
+                ? new StringBuilder("new " + functionSymbol.get().name().substring(0, initIndex))
+                : new StringBuilder(functionSymbol.get().name());
 
+        labelBuilder.append("(");
         // Join the function parameters to generate the function's signature
-        String paramsJoined = signatureInfoModel.getParameterInfoModels().stream().map(parameterInfoModel -> {
-            // For each of the parameters, create a parameter info instance
-            parameterInformationList.add(getParameterInformation(parameterInfoModel));
-
-            return parameterInfoModel.toString();
-        }).collect(Collectors.joining(", "));
+        List<ParameterInfoModel> parameterInfoModels = signatureInfoModel.getParameterInfoModels();
+        for (ParameterInfoModel paramModel : parameterInfoModels) {
+            int labelOffset = labelBuilder.toString().length();
+            labelBuilder.append(paramModel.parameter.getType());
+            ParameterInformation paramInfo = new ParameterInformation();
+            paramInfo.setDocumentation(getParameterDocumentation(paramModel));
+            int paramStart = labelOffset;
+            int paramEnd = labelOffset + paramModel.parameter.getType().length();
+            if (paramModel.parameter.getName().isPresent()) {
+                paramStart = paramEnd + 1;
+                paramEnd += (paramModel.parameter.getName().get() + " ").length();
+                labelBuilder.append(" ").append(paramModel.parameter.getName().get());
+            }
+            if (parameterInfoModels.indexOf(paramModel) < parameterInfoModels.size() - 1) {
+                labelBuilder.append(", ");
+            }
+            paramInfo.setLabel(Tuple.two(paramStart, paramEnd));
```

Comment:
can we move this logic related to setting `Tuple.two(paramStart, paramEnd)` into `getParameterInfoModels()`

Reply:
Same as the above and it will be better to keep the final model generation in once place. WDYT?

Reply:
Still, paramStart, paramEnd calculation can be a part of `ParameterInfoModel` of the `SignatureInfoModel`. WDYT?

Reply:
IMHO I think we don't need to, this is because the paramStart and paramEnd is calculated relative to the label. That's the reason for generating both the label and the params in a single run without calculating the parameters separately as per the earlier logic.

Reply:
Yeah agree, seems paramStart requires other labelling positions as input.